packages/block-editor/src/hooks/utils.js

```diff
@@ -516,6 +516,16 @@ export function createBlockListBlockFilter( features ) {
 				<BlockListBlock
 					key="edit"
 					{ ...props }
+					__unstableLayoutClassNames={ allWrapperProps
```

Comment:
This won't always be rendered because some blocks are returning early above when they don't have set values in their attributes. You can test this by inserting a Gallery and a Navigation block on a page: they won't have their proper layout because the layout classnames aren't being output.

Layout always outputs default styles, even if no attributes are set on the block. The styles are output based on the default layout type declared in the block.json. You only get layout related attributes explicitly declared on the block if you change the default settings for a layout type, e.g. change the justification or orientation on a flex block.

So I think for this to work the logic above should maybe check only what the block supports instead of its explicit attributes.

Reply:
Good catch, yep, it seems layout classes aren't being output on a simple Gallery block for me:

| Trunk | This PR |
| --- | --- |
| ![image](https://github.com/WordPress/gutenberg/assets/14988353/3b76ea7e-cd5d-4c3d-b8c2-ba800496a054) | ![image](https://github.com/WordPress/gutenberg/assets/14988353/a2035fa0-c30f-46b3-8c73-0c5e5b586db5) |